moveit_kinematics/ikfast_kinematics_plugin/templates/ikfast61_moveit_plugin_template.cpp

```diff
@@ -1060,13 +1093,41 @@ bool IKFastKinematicsPlugin::getPositionIK(const geometry_msgs::Pose& ik_pose, c
     ROS_DEBUG_STREAM_NAMED("ikfast", "No IK solution");
   }
 
+  // Sort the solutions under limits and find the one that is closest to ik_seed_state
```

Comment:
i believe this is bubble sort O(n^2)... the worse case performance? how about wrap the ``dists`` and ``solutions_obey_limits`` into a common ``struct``, then use [std::sort](http://en.cppreference.com/w/cpp/algorithm/sort) for a clean and optimal O(nlogn)?